frontend/src/modules/EditZing/Components/EditZing.tsx

```diff
@@ -165,9 +173,15 @@ export const EditZing = () => {
       .post(`${API_ROOT}${MATCHING_API}/make`, { courseId: courseId })
       .then((response) => {
         let newGroups = studentGroups.concat(response.data.data.groups)
-        console.log(response.data.data.unmatched)
+        newGroups = newGroups.map((group: Group) => {
+          return {
+            ...group,
+            shareMatchEmailTimestamp: group.shareMatchEmailTimestamp
+              ? new Date(group.shareMatchEmailTimestamp)
+              : null,
+          }
+        })
```

Comment:
I think newGroups might be a misnomer because in the line above it is actually `studentGroups.concat(response.data.data.groups)`. Coincidentally it still works because `new Date()` can take in an existing Date object, but we could make it clearer by only converting the new response data into Date objects.

Reply:
Yeah I realize that now, I changed it so it only maps with the returned groups response.